### ISSUE TITLE
What's the hyper-parameters for transfer learning on SentEval transfer tasks?

### ISSUE THREAD

**Issue by yaoxingcheng** [UKPLab/sentence-transformers#749]
Hi, I tried to reproduce the results for 7 transfer tasks in SentEval (i.e. MR, CR, SUBJ, MPQA, SST2, TREC and MRPC) selected in the original Sentence-BERT paper, using SentEval toolkit. I chose the following parameter setting in SentEval:
`params = {'task_path': PATH_TO_DATA, 'usepytorch': True, 'kfold': 10}
    params['classifier'] = {'nhid': 0, 'optim': 'adam', 'batch_size': 64,
                                     'tenacity': 5, 'epoch_size': 4}`
However, the results are generally worse than those reported in the original paper:
  | MR | CR | SUBJ | MPQA | SST2 | TREC | MRPC | Avg.
-- | -- | -- | -- | -- | -- | -- | -- | --
my implementation | 82.7 | 89.04 | 94.08 | 89.71 | 88.36 | 85.2 | 75.07 | 86.31
paper | 83.64 | 89.43 | 94.39 | 89.86 | 88.96 | 89.6 | 76 | 87.41

The pre-trained model is nli-bert-base downloaded from sbert.net, and the random seed is 1111 (default setting in SentEval). Would you mind elaborating on the hyper-parameter setting in Sentence-BERT paper for training the LR classifier to improve the reproducibility?

**Comment by nreimers** [comment]
I used the parameters they recommend in the readme:
```
params = {'task_path': PATH_TO_DATA, 'usepytorch': True, 'kfold': 10}
params['classifier'] = {'nhid': 0, 'optim': 'adam', 'batch_size': 64,
                                 'tenacity': 5, 'epoch_size': 4}
```

Note: Huggingface transformers was drastically changed since the SBERT paper was released. In the paper I used the original pytorch-bert-pre-trained, which is one of the first releases from Huggingface.

It might be that the vectors are now slightly different then what you would get with the old old old version of huggingface transformers / pytorch-bert-per-trained.
